src/lm_robust_helper.cpp

```diff
@@ -146,24 +146,24 @@ List lm_robust_helper(const arma::vec & y,
 
       if (type == "CR2") {
 
-        // if weights
+        // Instead of having X and Xt be weighted by sqrt(W), this has them weighted by W
         Xt = Xt.each_row() % arma::sqrt(arma::trans(weights));
+        X = X.each_col() % arma::sqrt(weights);
 
         arma::mat tutX(J, k);
 
-        arma::mat WU = arma::sqrt(weights) % X.each_col();
-        arma::mat M_U_ct = arma::trans(arma::chol(XtX_inv));
 
-        // Cube stores the n by k G_s matrix for each cluster
-        // used for the BM dof corrction
+        // used for the dof corrction
         arma::cube H1s(k, k, J);
         arma::cube H2s(k, k, J);
         arma::cube H3s(k, k, J);
         arma::mat P_diags(k, J);
 
-        ei = y / sqrt(weights) - (X.each_col() / sqrt(weights)) * beta_hat;
+        arma::mat M_U_ct = arma::trans(arma::chol(XtX_inv));
```

Comment:
chol has a `layout` argument, instead of doing a transpose. Not sure if arma is smart enough to optimize this away.

Reply:
Great, thanks.